charmcraft.yaml

```diff
@@ -31,40 +31,43 @@ parts:
       just requirements
       craftctl default
 
-subordinate: true
+charm-libs:
+  - lib: operator-libs-linux.apt
+    version: "0.15"
+  - lib: operator-libs-linux.systemd
+    version: "1.4"
 
-peers:
-  storage-peers:
-    interface: storage_peers
+subordinate: true
 
 requires:
-  fs-share:
-    interface: fs_share
+  filesystem:
+    interface: filesystem_info
   juju-info:
     interface: juju-info
     scope: container
 
 config:
   options:
-    mountinfo:
+    mounts:
       default: "{}"
       type: string
       description: |
-        Information to mount the filesystem on the machine. This is specified as a JSON object string.
+        Information to mount filesystems on the machine. This is specified as a JSON object string.
         Example usage:
           ```bash
-          $ juju config slurmd mountpoints=<<EOF
-            {
-              "cephfs": {
-                "mountpoint": "/scratch",
-                "noexec": true
-              },
-              "nfs": {
-                "mountpoint": "/data",
-                "nosuid": true,
-                "nodev": true,
-                "read-only": true,
-              }
-            }
-          EOF
+          $ juju config filesystem-client \
+              mounts=<<EOF
+                {
+                  "cephfs": {
+                    "mountpoint": "/scratch",
+                    "noexec": true
+                  },
+                  "nfs": {
+                    "mountpoint": "/data",
+                    "nosuid": true,
+                    "nodev": true,
+                    "read-only": true,
+                  }
+                }
+              EOF
```

Comment:
Per our in-person discussion, I think we should probably just have the filesystem-client support a single remote filesystem. If users want more filesystems, they can deploy additional applications.